web/react/components/create_comment.jsx

```diff
@@ -106,13 +106,23 @@ module.exports = React.createClass({
         }
     },
     handleUserInput: function(messageText) {
+        var newPostError = this.state.postError;
+
+        var len = utils.getLengthOfTextInTextarea(messageText);
+
+        if (!this.state.postError && len >= Constants.MAX_POST_LEN) {
+            newPostError = 'Comment length cannot exceed 4000 characters';
```

Comment:
In addition to crspeller's comment, you might want to use the MAX_POST_LEN constant for your error message instead of hardcoding the number.